src/components/ProjectSelector/index.js

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable no-script-url */
```

Comment:
I think you should try to use our ButtonMenu for this or at least our internal PrimitiveMenu component.
Remember that you can pass a component in the MenuItem label prop.
This would simplify the component dramatically because all this logic is already in PrimitiveMenu.